src/common/harvest/harvest.js

```diff
@@ -97,24 +101,28 @@ export class Harvest extends SharedContext {
 
     if (!opts) opts = {}
 
-    var url = this.getScheme() + '://' + info.errorBeacon + '/' + endpoint + '/1/' + info.licenseKey + this.baseQueryString()
-    if (payload.qs) url += encodeObj(payload.qs, agentRuntime.maxBytes)
+    var url = customUrl || this.getScheme() + '://' + info.errorBeacon + '/' + endpoint + '/1/' + info.licenseKey
 
+    var baseParams = includeBaseParams ? this.baseQueryString() : ''
+    var params = payload.qs ? encodeObj(payload.qs, agentRuntime.maxBytes) : ''
     if (!submitMethod) {
       submitMethod = getSubmitMethod(endpoint, opts)
     }
     var method = submitMethod.method
     var useBody = submitMethod.useBody
 
     var body
-    var fullUrl = url
-    if (useBody && endpoint === 'events') {
-      body = payload.body.e
-    } else if (useBody) {
-      body = stringify(payload.body)
-    } else {
-      fullUrl = url + encodeObj(payload.body, agentRuntime.maxBytes)
-    }
+    var fullUrl = url + '?' + baseParams + params
+
+    if (!gzip) {
```

Comment:
Compression shouldn't have anything to do with the content of the request body?

Reply:
outdated, will be changed